integration-tests/src/integration-tests/java/oracle/kubernetes/operator/BaseTest.java

```diff
@@ -0,0 +1,201 @@
+package oracle.kubernetes.operator;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import oracle.kubernetes.operator.utils.Domain;
+import oracle.kubernetes.operator.utils.Operator;
+import oracle.kubernetes.operator.utils.PersistentVolume;
+import oracle.kubernetes.operator.utils.Secret;
+import oracle.kubernetes.operator.utils.TestUtils;
+
+/**
+ * Base class which contains common methods to create/shutdown operator and domain. IT tests can
+ * extend this class.
+ *
+ * @author Vanajakshi Mukkara
+ */
+public class BaseTest {
+  protected static final String TESTWEBAPP = "testwebapp";
+
+  public static final Logger logger = Logger.getLogger("OperatorIT", "OperatorIT");
+
+  protected static Operator operator;
+  protected static Domain domain;
+  protected static PersistentVolume pv;
+  protected static Secret secret;
+
+  protected static String username = "weblogic";
+  protected static String password = "welcome1";
+
+  protected static String resultRoot = "";
+  protected static String pvRoot = "";
+  protected static String resultDir = "";
+  protected static String userProjectsDir = "";
+  protected static String pvDir = "";
+
+  protected static String opPropsFile = "";
+  protected static String domainPropsFile = "";
+  protected static String appPropsFile = "";
+  protected static Properties opProps = new Properties();
+  protected static Properties domainProps = new Properties();
+  protected static Properties appProps = new Properties();
+
+  protected static String operatorNS = "";
+  protected static String domainUid = "";
+  protected static String domainNS = "";
+
+  public static void setup() throws Exception {
+    boolean createOpDomain = true; //this flag will be removed later, its here for testing
+
+    //check file exists
+    File f = new File(BaseTest.class.getClassLoader().getResource(appPropsFile).getFile());
+    if (!f.exists()) {
+      throw new IllegalArgumentException(
+          "FAILURE: Invalid operator appp properties file " + appPropsFile);
```

Comment:
Can you re-use any of the operator test utils dealing with inputs files and output directories?
e.g. operator/src/test/java/oracle/kubernetes/operator/utils/UserProjects.java ?

Or are you not able to use them because you're under a different root directory (integration/src/...)?